### PR TITLE
Dev 10

### PR DESCRIPTION
**Pull request by akshaym99**
# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: scripts/cicd_k8s/client_multinode_rgw.py [hunk 1]
```diff
@@ -124,19 +124,33 @@ def main():
...
-
-    resp = ext_lb.configure_nodeport_lb(node_obj, "eth1")
-    if not resp[0]:
-        print("Did not get expected response: {}".format(resp))
-    ext_ip = resp[1]
-    port = resp[2]
-    ext_port_ip = "{}:{}".format(ext_ip, port)
-    print("External LB value, ip and port will be: {}".format(ext_port_ip))
+    if ext_node == "NodePort":
+        resp = ext_lb.configure_nodeport_lb(node_obj, "eth1")
```
Comment (pragamj) [inline scripts/cicd_k8s/client_multinode_rgw.py:131]: keep eth1 in config file and read it from config

File: commons/utils/ext_lbconfig_utils.py [hunk 2]
```diff
@@ -236,3 +236,88 @@ def configure_nodeport_lb(node_obj: LogicalNode, iface: str):
...
+    m_node_obj = LogicalNode(hostname=m_node, username=username, password=password)
+    resp = m_node_obj.execute_cmd(cmd=cm_cmd.K8S_WORKER_NODES, read_lines=True)
+    worker_node = {resp[index].strip("\n"): dict() for index in range(1, len(resp))}
+    resp = m_node_obj.execute_cmd(cmd=cm_cmd.CMD_GET_IP_IFACE.format("eth1"), read_lines=True)
+    master_eth1 = resp[0].strip("\n")
+    print("master eth1: ", master_eth1)
+    for worker in worker_node.keys():
+        w_node_obj = LogicalNode(hostname=worker, username=username, password=password)
+        resp = w_node_obj.execute_cmd(cmd=cm_cmd.CMD_GET_IP_IFACE.format("eth1"), read_lines=True)
+        worker_node[worker].update({"eth1": resp[0].strip("\n")})
```
Comment (sonalk0209) [inline commons/utils/ext_lbconfig_utils.py:258]: can we name eth1 with iface maybe to be generic?

**Review by pragamj — APPROVED** [review]

File: commons/utils/ext_lbconfig_utils.py [hunk 3]
```diff
@@ -236,3 +236,89 @@ def configure_nodeport_lb(node_obj: LogicalNode, iface: str):
...
+                get_iosvc_data[svc].update({iface: master_eth1})
+            else:
+                get_iosvc_data[svc].update({iface: worker_eth1.pop()})
+            if item_data["spec"].get("ports") is not None:
+                for port_items in item_data["spec"]["ports"]:
+                    get_iosvc_data[svc].update({f"{port_items['targetPort']}":
+                                                    port_items["nodePort"]})
+            else:
+                LOGGER.info("Failed to get ports details from %s", get_iosvc_data.get(svc))
+    print("get_iosvc_data is: ", get_iosvc_data)
```
Comment (swapnil-khandare) [inline commons/utils/ext_lbconfig_utils.py:281]: use logger

**Review by swapnil-khandare — APPROVED** [review]

**Review by sonalk0209 — APPROVED** [review]

**Comment by akshaym99** [comment]
pytest collection is green & successful

 'tests/s3/versioning/test_versioning_put_object.py::TestVersioningPutObject::test_put_object_versioning_suspended_32733', 'test_id': 'TEST-32733', 'marks': ['s3_ops']}, {'nodeid': 'tests/security/test_cortx_port_scanner_kubectl_svc.py::test_cortx_port_scanner_kubectl_svc', 'test_id': 'TEST-34217', 'marks': ['security']}, {'nodeid': 'tests/security/test_cortx_port_scanner_netstat.py::test_cortx_port_scanner_netstat', 'test_id': 'TEST-34218', 'marks': ['security']}] created at /root/akshay/cortx-test/log/te_meta.json
Successfully unmounted directory

================================================================================ 2036 tests collected in 11.71s =================================================================================
(virenv) [root@ssc-vm-g4-rhev4-0125 cortx-test]#

**Pull request merged**